src/libproxychains.c

```diff
@@ -587,6 +600,51 @@ static int is_v4inv6(const struct in6_addr *a) {
 	return !memcmp(a->s6_addr, "\0\0\0\0\0\0\0\0\0\0\xff\xff", 12);
 }
 
+static int compare_func_int(const void *l, const void *r) {
+    int arg1 = *(const int*)l;
+    int arg2 = *(const int*)r;
+    return (arg1 > arg2) - (arg1 < arg2);
+}
+/* Warning: Linux manual says the third arg is `unsigned int`, but unistd.h says `int`. */
+HOOKFUNC(int, close_range, unsigned first, unsigned last, int flags) {
+    if(!init_l) {
+        /* push back to cache, and delay the execution. */
+        if(close_range_buffer_cnt >= (sizeof close_range_buffer / sizeof *close_range_buffer)) {
+            errno = ENOMEM;
+            return -1;
+        }
+        int i = close_range_buffer_cnt++;
+        close_range_buffer[i].first = first;
+        close_range_buffer[i].last = last;
+        close_range_buffer[i].flags = flags;
+        return errno = 0;
+    }
```

Comment:
there are various whitespace bugs like this one here using spaces instead of tabs

Reply:
TODO: will fix this after fixing other issues